src/System.Private.CoreLib/shared/System/Memory.cs

```diff
@@ -209,9 +204,9 @@ public override string ToString()
         {
             if (typeof(T) == typeof(char))
             {
-                return (_object is string str) ? str.Substring(_index, _length & RemoveFlagsBitMask) : Span.ToString();
+                return (_object is string str) ? str.Substring(_index, _length) : Span.ToString();
```

Comment:
Not necessary for this PR, but I wonder if there's a real benefit to using Substring here, or whether this could just be:
```C#
if (typeof(T) == typeof(char))
{
    return Span.ToString();
}
```

Reply:
`Substring`'s primary benefit is that it will return the original string instance if you pass `(0, Length)` as inputs. We could do something similar:

```cs
if (typeof(T) == typeof(char)) {
  if (_object is string str && ((int index = _index) == 0) && ((int length = _length & RemoveFlagsBitmask) == str.Length)) {
    return str;
  } else {
    // somehow rewrite the below line so it's not duplicating the type / index / length checks we just performed above
    return Span.ToString();
  }
}
```

But we're complicating the call site, so it seems unnecessary.

Reply:
> Substring's primary benefit is that it will return the original string instance if you pass (0, Length) as inputs

Yup, good point.